### PR TITLE
cdc: merge two-phase scheduler

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve?

Issue Number: ref #4757 

### What is changed and how it works?

Add two-phase scheduler to reduce latency spike during rolling restart.

Commits are cherry-picked from [fb/latency](https://github.com/pingcap/tiflow/tree/fb/latency) branch.

#### Tests 

 - Unit test

### Release note 

```release-note
Add two-phase scheduler to reduce latency spike during rolling restart.
```

### CODE REVIEW

**Comment by amyangfei** [comment]
Should it be `tide/merge-method-merge` instead of `tide/merge-method-rebase`?

**Comment by overvenus** [comment]
> Should it be `tide/merge-method-merge` instead of `tide/merge-method-rebase`?

Both are fine, they all keep commit history, while rebase keep a linear history.

**Review by 3AceShowHand — APPROVED** [review]

File: cdc/api/v1/api.go [hunk 1]
```diff
@@ -695,6 +696,68 @@ func (h *OpenAPI) ListCapture(c *gin.Context) {
...
+// @Summary Drain captures
+// @Description Drain all tables at the target captures in cdc cluster
+// @Tags capture
+// @Accept json
+// @Produce json
+// @Success 200,202
+// @Failure 500,400 {object} model.HTTPError
+// @Router	/api/v1/captures/drain [post]
+func (h *OpenAPI) DrainCapture(c *gin.Context) {
+	if !h.capture.IsOwner() {
```
Comment (lonng) [inline cdc/api/v1/api.go:709]: We use middleware to forward the traffic to the owner and will the path reachable?

File: scripts/check-diff-line-width.sh [hunk 2]
```diff
@@ -23,7 +23,10 @@ BASE_HASH=$(git --no-pager log -E --grep='\(#[0-9]+\)$' -n 1 --format=format:%H)
 WARN_THRESHOLD=80
 ERROR_THRESHOLD=100
 
-git --no-pager diff $BASE_HASH -U0 -- cdc pkg cmd -- ':(exclude)*_gen.go' -- ':(exclude)*_gen_test.go' |
+git --no-pager diff $BASE_HASH -U0 -- cdc pkg cmd \
+	-- ':(exclude)*_gen.go' \
+	-- ':(exclude)*_gen_test.go' \
+	-- ':(exclude)*.pb.go' |
```
Comment (sdojjy) [inline scripts/check-diff-line-width.sh:29]: exclude *_mock.go  ?

Comment (overvenus) [reply scripts/check-diff-line-width.sh:29]: I prefer exclude them in another PR.

File: pkg/config/scheduler_config.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+import (
+	"time"
+
+	cerrors "github.com/pingcap/tiflow/pkg/errors"
+)
+
+// SchedulerConfig configs TiCDC scheduler.
+type SchedulerConfig struct {
+	HeartbeatTick        int          `toml:"heartbeat-tick" json:"heartbeat-tick"`
```
Comment (sdojjy) [inline pkg/config/scheduler_config.go:24]: from the field name`HeartbeatTick`, I don't know what it means.

**Review by sdojjy — APPROVED** [review]

File: pkg/config/scheduler_config.go (hunk 3 above)
Comment (overvenus) [reply pkg/config/scheduler_config.go]: Comments added, thanks!

**Pull request merged**